lib/jsdom/living/geometry/DOMMatrixReadOnly-impl.js

```diff
@@ -0,0 +1,519 @@
+
+    m.translateSelf(tX, tY, tZ);
+
+    return m;
+  }
+
+  scale(scaleX = 1, scaleY, scaleZ = 1, originX = 0, originY = 0, originZ = 0) {
+    if (typeof scaleY === "undefined") {
+      scaleY = scaleX;
+    }
+
+    const m = DOMMatrix.createImpl([this._getCurrentMatrix()]);
+
+    m.scaleSelf(scaleX, scaleY, scaleZ, originX, originY, originZ);
+
+    return m;
+  }
+
+  scaleNonUniform(scaleX = 1, scaleY = 1) {
+    // Deprecated: https://www.w3.org/TR/geometry-1/#dom-dommatrixreadonly-scalenonuniform
+
+    const m = DOMMatrix.createImpl([this._getCurrentMatrix()]);
+
+    m.scaleSelf(scaleX, scaleY);
+
+    return m;
+  }
+
+  scale3d(scale = 1, originX = 0, originY = 0, originZ = 0) {
+    const m = DOMMatrix.createImpl([this._getCurrentMatrix()]);
+
+    m.scale3dSelf(scale, originX, originY, originZ);
+
+    return m;
+  }
+
+  rotate(rotX = 0, rotY, rotZ) {
+    const m = DOMMatrix.createImpl([this._getCurrentMatrix()]);
+
+    m.rotateSelf(rotX, rotY, rotZ);
+
+    return m;
+  }
+
+  rotateFromVector(x = 0, y = 0) {
+    const m = DOMMatrix.createImpl([this._getCurrentMatrix()]);
+
+    m.rotateFromVectorSelf(x, y);
+
+    return m;
+  }
+
+  rotateAxisAngle(x = 0, y = 0, z = 0, angle = 0) {
+    const m = DOMMatrix.createImpl([this._getCurrentMatrix()]);
+    m.rotateAxisAngleSelf(x, y, z, angle);
+
+    return m;
+  }
+
+  skewX(sX = 0) {
+    const m = DOMMatrix.createImpl([this._getCurrentMatrix()]);
+
+    m.skewXSelf(sX);
+
+    return m;
+  }
+
+  skewY(sY = 0) {
+    const m = DOMMatrix.createImpl([this._getCurrentMatrix()]);
+
+    m.skewYSelf(sY);
+
+    return m;
+  }
+
+  multiply(other) {
+    const m = DOMMatrix.createImpl([this._getCurrentMatrix()]);
+
+    m.multiplySelf(other);
+
+    return m;
+  }
+
+  flipX() {
+    const m = DOMMatrix.create([this._getCurrentMatrix()]);
+
+    m.multiplySelf(DOMMatrix.create([[-1, 0, 0, 1, 0, 0]]).toJSON());
+
+    return m;
+  }
+
+  flipY() {
+    const m = DOMMatrix.createImpl([this._getCurrentMatrix()]);
+
+    m.multiplySelf(DOMMatrix.createImpl([[1, 0, 0, -1, 0, 0]]).toJSON());
+
+    return m;
+  }
+
+  inverse() {
+    const m = DOMMatrix.createImpl([this._getCurrentMatrix()]);
+
+    m.invertSelf();
+
+    return m;
+  }
+
+  // transformPoint(point) {
+  //   throw new Error(`Transform Point is not implemented`);
+  // }
+
+  toFloat32Array() {
+    return new Float32Array(this._getCurrentMatrix());
+  }
+
+  toFloat64Array() {
+    return new Float64Array(this._getCurrentMatrix());
+  }
+
+  toJSON() {
+    return [
+      "a", "b", "c", "d", "e", "f",
+      "m11", "m12", "m13", "m14",
+      "m21", "m22", "m23", "m24",
+      "m31", "m32", "m33", "m34",
+      "m41", "m42", "m43", "m44",
+      "is2D", "isIdentity"
+    ].reduce((domMatrixInit, prop) => {
+      domMatrixInit[prop] = this[prop];
+      return domMatrixInit;
+    }, {});
+  }
+
+  toString() {
+    const matrix = this._getCurrentMatrix();
+
+    if (matrix.some(x =>
```

Comment:
```suggestion
    if (!matrix.every(Number.isFinite))
```

Generally prefer the `Number.is`… variants as they type-check the input argument.

Reply:
That's great, first I've seen the `Number.is` functions